jsonpb/jsonpb.go

```diff
@@ -264,7 +264,26 @@ func (m *Marshaler) marshalObject(out *errWriter, v proto.Message, indent, typeU
 			return m.marshalValue(out, &proto.Properties{}, x, indent)
 		}
 	}
-
+	// To correctly support marshaling field masks, the following is largely copied from the `golang/protobuf` v2 implementation
+	// https://github.com/golang/protobuf/blob/a3369c5dc28ac197f8ec8de40a7291be4dea85b5/encoding/protojson/well_known_types.go#L884
+	if proto.MessageName(v) == "google.protobuf.FieldMask" {
+		list := s.Field(0)
+		paths := make([]string, 0, list.Len())
+		for i := 0; i < list.Len(); i++ {
+			s := list.Index(i).String()
+			// Return error if conversion to camelCase is not reversible.
+			cc := JSONCamelCase(s)
+			if s != JSONSnakeCase(cc) {
+				return fmt.Errorf("paths contains irreversible value %q", s)
```

Comment:
Can we collect the errors here with `multierr`? With the current error logic, it would cut at the first error, and at a user's perspective, if they have multiple errors, they would have to run the compiler, fix the error, and then run the compiler again to see the next error.

Reply:
Keeping the error handling consistent with rest of the jsonpb and with the golang v2 implementation. multierr is not used.